samples/jpegDecodePerf/jpegdecodeperf.cpp

```diff
@@ -0,0 +1,353 @@
+    std::vector<std::string> valid_base_file_names(batch_size);
+    std::vector<int> bad_image_indices;
+    std::vector<RocJpegStreamHandle> valid_rocjpeg_stream_handles(batch_size);
+
+    for (int i = 0; i < file_paths.size(); i += batch_size) {
+        int batch_end = std::min(i + batch_size, static_cast<int>(file_paths.size()));
+        for (int j = i; j < batch_end; j++) {
+            int index = j - i;
+            base_file_names[index] = file_paths[j].substr(file_paths[j].find_last_of("/\\") + 1);
+            // Read an image from disk.
+            std::ifstream input(file_paths[j].c_str(), std::ios::in | std::ios::binary | std::ios::ate);
+            if (!(input.is_open())) {
+                std::cerr << "ERROR: Cannot open image: " << file_paths[j] << std::endl;
+                return;
+            }
+            // Get the size
+            std::streamsize file_size = input.tellg();
+            input.seekg(0, std::ios::beg);
+            // resize if buffer is too small
+            if (batch_images[index].size() < file_size) {
+                batch_images[index].resize(file_size);
+            }
+            if (!input.read(batch_images[index].data(), file_size)) {
+                std::cerr << "ERROR: Cannot read from file: " << file_paths[j] << std::endl;
+                return;
+            }
+
+            RocJpegStatus rocjpeg_status = rocJpegStreamParse(reinterpret_cast<uint8_t*>(batch_images[index].data()), file_size, rocjpeg_stream_handles[index]);
+            if (rocjpeg_status != ROCJPEG_STATUS_SUCCESS) {
+                bad_image_indices.push_back(index);
+                *num_bad_jpegs += 1;
+                std::cerr << "Skipping decoding input file: " << file_paths[j] << std::endl;
+                continue;
+            }
+
+            CHECK_ROCJPEG(rocJpegGetImageInfo(rocjpeg_handle, rocjpeg_stream_handles[index], &num_components, &subsamplings[index], widths[index].data(), heights[index].data()));
+
+            if (roi_width > 0 && roi_height > 0 && roi_width <= widths[index][0] && roi_height <= heights[index][0]) {
+                is_roi_valid = true; 
+            }
+
+            rocjpeg_utils.GetChromaSubsamplingStr(subsamplings[index], chroma_sub_sampling);
+            if (widths[index][0] < 64 || heights[index][0] < 64) {
+                bad_image_indices.push_back(index);
+                *num_jpegs_with_unsupported_resolution += 1;
+                continue;
+            }
+
+            if (subsamplings[index] == ROCJPEG_CSS_411 || subsamplings[index] == ROCJPEG_CSS_UNKNOWN) {
+                bad_image_indices.push_back(index);
+                if (subsamplings[index] == ROCJPEG_CSS_411) {
+                    *num_jpegs_with_411_subsampling += 1;
+                }
+                if (subsamplings[index] == ROCJPEG_CSS_UNKNOWN) {
+                    *num_jpegs_with_unknown_subsampling += 1;
+                }
+                continue;
+            }
+
```

Comment:
Instead of figuring out valid images in line#149 onwards, you can push those in appropriate vectors here. You can dynamically grow the vectors here and avoid checking for those again in line #154. Instead of fixed size vectors, you can push each valid entry into the appropriate vectors

Reply:
Done. I modified the logic for valid image selection.